### PR TITLE
Remove subprocess in backfill_agent, and support CSV output for query_data_from_perfherder

### PR DESCRIPTION
**Pull request by askeing**

### CODE REVIEW

**Comment by askeing** [comment]
@ShakoHo @ypwalter @MikeLien @ycyen r?

**Comment by askeing** [comment]
@ycyen please help to verify this patch, thanks :)

**Comment by ypwalter** [comment]
I will verify this patch by applying it in today's machine. All machine had the same leaking without this patch.

File: tools/query_data_from_perfherder.py [hunk 1]
```diff
@@ -123,44 +129,122 @@ def print_data(self, input_json, signature_data, b_date, e_date):
...
+            for sig in input_json:
+                for data in input_json[sig]:
+                    if b_timestamp != 0.0:
+                        if e_timestamp != 0.0:
+                            if b_timestamp <= data['push_timestamp'] <= e_timestamp:
+                                suite_name_full = signature_data['signature_data'][sig]['suite_name'].split()
+                                suite_name = suite_name_full[0]
+                                suite_name_type = suite_name_full[1]
+                                data_date = datetime.utcfromtimestamp(data['push_timestamp']).strftime('%Y-%m-%d')
+                                data_time = datetime.utcfromtimestamp(data['push_timestamp']).strftime('%H:%M:%S.%f')
```
Comment (ypwalter) [inline tools/query_data_from_perfherder.py]: Should %H-%M-%S-%f

File: tools/query_data_from_perfherder.py [hunk 2]
```diff
@@ -123,44 +129,122 @@ def print_data(self, input_json, signature_data, b_date, e_date):
...
+                                                     data_date,
+                                                     data_time,
+                                                     data['value']])
+                        else:
+                            if b_timestamp <= data['push_timestamp']:
+                                suite_name_full = signature_data['signature_data'][sig]['suite_name'].split()
+                                suite_name = suite_name_full[0]
+                                suite_name_type = suite_name_full[1]
+                                data_date = datetime.utcfromtimestamp(data['push_timestamp']).strftime('%Y-%m-%d')
+                                data_time = datetime.utcfromtimestamp(data['push_timestamp']).strftime('%H:%M:%S.%f')
```
Comment (ypwalter) [inline tools/query_data_from_perfherder.py]: Should %H-%M-%S-%f

File: tools/query_data_from_perfherder.py [hunk 3]
```diff
@@ -123,44 +129,122 @@ def print_data(self, input_json, signature_data, b_date, e_date):
...
+                                                     data_time,
+                                                     data['value']])
+                    else:
+                        if e_timestamp != 0.0:
+                            if data['push_timestamp'] <= e_timestamp:
+                                suite_name_full = signature_data['signature_data'][sig]['suite_name'].split()
+                                suite_name = suite_name_full[0]
+                                suite_name_type = suite_name_full[1]
+                                data_date = datetime.utcfromtimestamp(data['push_timestamp']).strftime('%Y-%m-%d')
+                                data_time = datetime.utcfromtimestamp(data['push_timestamp']).strftime('%H:%M:%S.%f')
```
Comment (ypwalter) [inline tools/query_data_from_perfherder.py]: Should %H-%M-%S-%f

**Comment by ypwalter** [comment]
It is been verify in most windows 7 and windows 10 machines. Let's see how it goes.

**Pull request merged**